src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -790,7 +790,16 @@ DistributedTableCacheEntry(Oid distributedRelationId)
 	else
 	{
 		char *relationName = get_rel_name(distributedRelationId);
-		ereport(ERROR, (errmsg("relation %s is not distributed", relationName)));
+
+		if (relationName == NULL)
```

Comment:
To keep things simpler, what about:

```C
/* should never be NULL, but it doesn't hurt to be paranoid */
relationName = relationName ? relationName : "null";
ereport(ERROR, (errmsg("relation %s is not distributed", relationName)));
```

Reply:
Printing relationName="null" for a table that doesn't exist doen't sound me like an intuitive one actually

Reply:
we normally don't expect anything like that to happen ever  (remember it was an assert in your first commit), it is just to prevent crashes in case we forgot some weird edge cases with drop table, like the one you hit earlier such as #3526.

The error message `relation with OID %u does not exist` is confusing to me. The question that pops in my head is If it doesn't exist, how do we see this error :) 

"null" is also not very clear, but it at least gives the user that Citus somehow wasn't able to find the relation name.

And, what bugs me in this change is having an `if/else` for such a simple and unlikely event.

Anyway, feel free to leave as-is if you think that's better.

Reply:
Seeing the approval, I had already merged the pr. Sorry for that :/